models/Search/Backend/Data.php

```diff
@@ -507,7 +507,7 @@ public function setDataFromElement($element)
      */
     protected function cleanupData($data)
     {
-        $data = strip_tags($data);
+        $data = preg_replace('/(<\?.*?(\?>|$)|<.*?>)/', '', $data);
```

Comment:
Maybe we need the `s` modifier here, in case the content exceeds one line.

```suggestion
        $data = preg_replace('/(<\?.*?(\?>|$)|<.*?>)/s', '', $data);
```